src/compiler/legend.ts

```diff
@@ -43,7 +43,7 @@ export function compileLegend(model: Model, channel: Channel, def) {
   });
 
   // 2) Add mark property definition groups
-  let props = legend.properties || {};
+  const props = typeof legend === 'Legend' && legend.properties || {};
```

Comment:
This line is terrible!  Nobody on earth remember precedence of these two operators. 
I also do not get what you try to do here?